src/test/ui/parser/issue-44406.stderr

```diff
@@ -9,17 +9,25 @@ help: you can escape reserved keywords to use them as identifiers
 LL |     foo!(r#true);
    |          ~~~~~~
 
-error: expected type, found keyword `true`
-  --> $DIR/issue-44406.rs:8:10
+error: invalid `struct` delimiters or `fn` call arguments
+  --> $DIR/issue-44406.rs:3:9
    |
 LL |         bar(baz: $rest)
-   |                - help: try using a semicolon: `;`
+   |         ^^^^^^^^^^^^^^^
 ...
 LL |     foo!(true);
-   |          ^^^^ expected type
+   |     ----------- in this macro invocation
+   |
+   = note: this error originates in the macro `foo` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: if `bar` is a struct, use braces as delimiters
+   |
+LL |         bar {  }
```

Comment:
This seems messed up. Shouldn't it be `bar { baz: $rest }` ?

Reply:
This is indeed "not correct", caused by the macro context of the spans involved. I could _never_ show the suggestions if a macro is involved, but that would hide the suggestion even if the macro _isn't_ mixing fragments coming from the caller with internal spans. Then again a middle point would be to give just a `help` without a suggestion for things in macros now that I think about it 🤔